base/dict.jl

```diff
@@ -111,6 +111,9 @@ mutable struct Dict{K,V} <: Associative{K,V}
         new(copy(d.slots), copy(d.keys), copy(d.vals), 0, d.count, d.age, d.idxfloor,
             d.maxprobe)
     end
+    function Dict{K, V}(slots, keys, ndel, count, age, idxfloor, maxprobe) where {K, V}
```

Comment:
FYI I left this constructor in so I might continue to experiment with `similar` and tokens and indexing and so-on outside of `Base`. Happy to remove, or change this to a simple one that also lets one specify `values`, as people wish.